src/nidigital/examples/burst_with_programmatic_configuration/nidigital_burst_with_programmatic_configuration.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python
+
+import argparse
+import nidigital
+import os
+import sys
+
+
+class VoltageLevelsAndTerminationConfig():
+    def __init__(self, vil, vih, vol, voh, vterm, termination_mode, iol, ioh, vcom):
+        self.vil = vil
+        self.vih = vih
+        self.vol = vol
+        self.voh = voh
+        self.vterm = vterm
+        self.termination_mode = termination_mode
+        self.iol = iol
+        self.ioh = ioh
+        self.vcom = vcom
+
+
+class TimeSetConfig():
+    def __init__(self, time_set_name, period, drive_format, drive_on, drive_data, drive_return, drive_off, strobe_edge):
+        self.time_set_name = time_set_name
+        self.period = period
+        self.drive_format = drive_format
+        self.drive_on = drive_on
+        self.drive_data = drive_data
+        self.drive_return = drive_return
+        self.drive_off = drive_off
+        self.strobe_edge = strobe_edge
+
+
+def convert_drive_format(drive_format):
+    converter = {'NR': nidigital.DriveFormat.NR,
+                 'RL': nidigital.DriveFormat.RL,
+                 'RH': nidigital.DriveFormat.RH,
+                 'SBC': nidigital.DriveFormat.SBC}
+    return converter.get(drive_format, None)
+
+
+def example(resource_name, options,
+            command,
+            channels=None,
+            voltage_config=None,
+            time_set_config=None):
+
+    with nidigital.Session(resource_name=resource_name, options=options) as session:
+
+        dir = os.path.dirname(__file__)
+
+        # Load pin map (.pinmap) created using Digital Pattern Editor
+        pin_map_filename = os.path.join(dir, 'PinMap.pinmap')
+        session.load_pin_map(pin_map_filename)
+
+        # Load the specifications (.specs), levels (.digilevels), and timing (.digitiming) sheets created using Digital Pattern Editor
+        spec_filename = os.path.join(dir, 'Specifications.specs')
+        levels_filename = None if command == 'configure-voltage' else os.path.join(dir, 'PinLevels.digilevels')
+        timing_filename = None if command == 'configure-time-set' else os.path.join(dir, 'Timing.digitiming')
+
+        session.load_specifications_levels_and_timing(spec_filename, levels_filename, timing_filename)
+
+        # Apply the settings from the levels and timing sheets we just loaded
+        session.apply_levels_and_timing(levels_filename if levels_filename is not None else '', timing_filename if timing_filename is not None else '')
```

Comment:
I think we should demonstrate loading the file and calling apply only if they choose not to configure using low-level API. Else it would be confusing.

So, either configure via files or via low-level API, depending on the configuration arg they pass in.